examples/with-react-intl/components/PageWithIntl.js

```diff
@@ -0,0 +1,44 @@
+import React, {Component} from 'react'
+import {IntlProvider, addLocaleData, injectIntl} from 'react-intl'
+
+// Register React Intl's locale data for the user's locale in the browser. This
+// locale data was added to the page by `pages/_document.js`. This only happens
+// once, on initial page load in the browser.
+if (typeof window !== 'undefined' && window.ReactIntlLocaleData) {
```

Comment:
Sure I don't challenge that the PR is valid, it's a progress coming from nothing 😄 We can do what we want with these comments, it's just that I wouldn't want someone to discover the whole subject of next + react-intl this way, it doesn't exactly simplify the DX (which is hopefully the main purpose of Next.js).

1. how is using `eval` worse than `dangerouslySetInnerHtml`? Both sound ominous to me anyway. (but I'm no expert on this)
2. I'm referring to the fact that i18n needs to be spread in 3 main files (PageWithIntl.js + _document.js + server.js, which normally isn't made for custom APIs, BTW 😄). I18n is a single cross-site concern, and as such, I'd IMHO strive to keep it in a single HOC (and if not possible, see how we can make it happen).

Currently for my app I just took the middle path of manually loading all localeData, even client-side. I hope not to spread i18n logic in my custom server etc just to handle this. But I do load the messages from the custom server... https://github.com/relatenow/relate

Agreed using serialize-javascript sounds like a good thing to investigate, maybe you can create an issue or should I?